tests/test_download.py

```diff
@@ -0,0 +1,62 @@
+from tests import OkTestCase
+
+from server.models import db, Assignment, Backup, Message, User
+from server import utils
+
+class TestDownload(OkTestCase):
+    def _add_file(self, filename, contents):
+        self.setup_course()
+        
+        email = 'student1@okpy.org'
+        self.login(email)
+        self.user = User.lookup(email)
+        
+        self.backup = Backup(
+            submitter=self.user,
+            assignment=self.assignment,
+            submit=True)
+            
+        self.message = Message(
+            backup=self.backup,
+            contents={
+                filename: contents,
+                'submit': True
+            },
+            kind='file_contents')
+            
+        db.session.add(self.backup)
+        db.session.add(self.message)
+        db.session.commit()
+        
+    def test_simple(self):
+        filename = "test.py"
+        contents = "x = 4"
+        self._add_file(filename, contents)
+        encoded_id = utils.encode_id(self.backup.id)
```

Comment:
For future cleanup: seems like we can generalize this code (34-40) to a `_download` helper since it's repeated.